ui/src/PrimerPage.tsx

```diff
@@ -18,6 +18,7 @@ interface Props {
   annotationHintsEnabled: boolean;
   termGlossesEnabled: boolean;
   scrollToPageOnLoad?: boolean;
+  isCitationsLoading?: boolean;
```

Comment:
nit pick: maybe `areCitationsLoading` / `isLoadingCitations`?

Reply:
Another thought is that below you've got a callback called 'isEntitiesLoaded' so maybe this could be  called 'isCitationsLoaded' (flagging completion rather than progress)

Reply:
the function there is a type guard, so it returns "this thing is of type Entities" (which is how we determine whether or not the entities are loaded). For the boolean, my preference is `true` for loading states and `false` in the (more normal) loaded state.